### ISSUE TITLE
CustomTooltip on Scatter is not displayed on 1st visit to point.

### ISSUE THREAD

**Issue by StuartDeanBaillie** [recharts/recharts#3547]
I have had a search and could not find a duplicate issue.

I have been tasked with looking up additional data for my customtooltip.  When I fake the extra data directly in the customtooltip, or provide the extra data in the payload, the customtooltip always is displayed on 1st hover.  If the extra data is faked or collected from a database within a useQuery hook, the customtooltip is not displayed on 1st hover..  but is always displayed on subsequent visits until it is expired from cache. 

I have tried several settings incase it was related to animations.   I have also tried using state in the customtooltip to see if it would trigger a re-render, but as others have remarked,  using state in a customtooltip causes infinite recursion.

I will attempt to create an example using the react example for in the react recharts source, to see if it being dismissed before being displayed..

**Comment by StuartDeanBaillie** [comment]
I have a feeling this is because useQuery is re-rendering the component,  which has then lost its on hover state,  but is written to cache so the following on hover doesn't re-render instead using the previously created component...  is there a workaround to this?

**Comment by ckifer** [comment]
Hey @StuartDeanBaillie - its hard to imagine the setup you have going here without an example (even if contrived). 

It seems you are making a network call on each hover within a custom tooltip?

**Comment by nikolasrieble** [comment]
@StuartDeanBaillie Please do provide a minimal reproducible example, then we can reproduce, investigate the bug, and share a fix.

**Comment by StuartDeanBaillie** [comment]
Thank you for you prompt responses, I will try and use the online tools, people have used in things like code sandbox, to create a simple example. 

But yes,  I am making a network call to retrieve lots of extra data for the tooltip, so the chart is populated with just x,y,unit and vehicle identifier in the payload...  then I use the vehicle identifier to bring back lots of information..   this is instead of including all the extra details in the data when populating the chart..  

I have found similar issues raised from  up to 4 or 5 years back,  surrounding when any of the state was changed the tooltip would disappear - there was maybe 7 or 8 issues linked to a fix which was made about 3 years ago...  

There maybe a simple workaround, like knowing what the last tool tip was, and setting the tooltip back to active..   

I will make sure my manager is happy for me to continue with this issue today,  if not I will continue checking after work this evening..

**Comment by StuartDeanBaillie** [comment]
Hi,

I have a reproducible version on codesandbox now.  it is faking data for a vehicle - if you hover a data point the cross hair appears without the tooltip,  then leave the data point and return and the custom tooltip will appear,  the data expires after about 5 minutes.

thanks for any work-around / fix 

https://codesandbox.io/s/dawn-wind-nwb9v0

**Comment by StuartDeanBaillie** [comment]
Okay so the issues have been discovered..  two changes were needed in the code sandbox version I didn't return something when tool tip not active (although that was not the case in my actual app.  but the useQuery needed to have 
**enabled: !!identificationNumber** 
rather than return null in the actualfn..

I have learnt a lot though :)

**Comment by StuartDeanBaillie** [comment]
example after throttling chrome to **slow 3g** in my application..  it shows my custom tooltip default return.  

![image](https://user-images.githubusercontent.com/27852652/234030356-81011d85-92a4-4d76-9357-ec4a171e7ff2.png)

before then updating the content from the Mongo database on Azure, but not refreshing the the position, a second later.

**Comment by ckifer** [comment]
sorry for not getting back to this - the example is still a bit hard to follow but I see what you're saying. Labelled
